src/lib/xmlparse.rs

```diff
@@ -879,19 +876,19 @@ macro_rules! hash_lookup {
 #[repr(C)]
 #[derive(Clone)]
 pub struct DTD {
-    pub generalEntities: HashMap<HashKey, ExpatBox<ENTITY>>,
-    pub elementTypes: HashMap<HashKey, ExpatBox<ELEMENT_TYPE>>,
-    pub attributeIds: HashMap<HashKey, ExpatBox<ATTRIBUTE_ID>>,
-    pub prefixes: HashMap<HashKey, ExpatBox<PREFIX>>,
+    pub generalEntities: HashMap<HashKey, Box<ENTITY>>,
```

Comment:
I would reeeeeeeeeeeeally like to get rid of these boxes, but tests start failing if I remove them (even now). On top of that, this is also what the original C code does: each hash table value is a pointer to a separate allocation.

Reply:
We should probably wait with optimizations until we're done making such massive changes all the time and have mostly safe code. Maybe add a TODO so we don't forget?